pro/src/pages/Signup/SignupContainer/SignupContainer.tsx

```diff
@@ -37,9 +37,8 @@ export const SignupContainer = (): JSX.Element => {
       if (response === RECAPTCHA_ERROR) {
         notification.error(RECAPTCHA_ERROR_MESSAGE)
       } else {
-        // TODO type this
-        // @ts-expect-error
-        onHandleFail(response.body ? response.body : {})
+        const body = (response as ApiError).body
+        onHandleFail(body ?? {})
```

Comment:
Ça risque de casser si l'erreur est pas `ApiError`.
Et pas sûr que le `@ts-expect-error` était pour ça, non ?
```suggestion
        const body = isErrorAPIError(response) ? response.body : {}
        onHandleFail(body)
```

Reply:
Le `@ts-expect-error` était bien là pour ça. Pour en déduire le `as ApiError`, (et n'ayant pas connaissance de la fonction `isErrorAPIError`), j'avais été rechercher manuellement quel type d'exception pouvait être renvoyé par le bloc "try" (`RECAPTCHA_ERROR` ou `ApiError`). Comme la condition juste au dessus prenait déjà en cas de figure le RECAPTCHA_ERROR, j'en ai déduit qu'après ce serait forcément une `ApiError`.